Clustering/Hdbscan/src/main/java/org/tribuo/clustering/hdbscan/HdbscanTrainer.java

```diff
@@ -762,6 +724,32 @@ private static double getDistance(SGDVector vector1, SGDVector vector2, Distance
         return distance;
     }
 
+    /**
+     * This method performs a mapping from a {@link Distance} to the corresponding {@link DistanceType}.
+     * @deprecated This method is deprecated in version 4.3.
+     *
+     * @param distance The distance metric to employ.
+     * @return A {@link DistanceType} value mapped from the provided parameter.
+     */
+    @Deprecated
+    private static DistanceType getDistanceType(Distance distance) {
```

Comment:
I think for the actual integration it would be better to add the `DistanceType` as a field on the `Distance` enum which can be accessed by a getter. That'll remove the extra switch. Then alter the signature of the private methods to accept `DistanceType` rather than `Distance`, and push the conversion up as high as we can (maybe even into postConfig on the trainer, add distance type as a configurable field and throw if both are set).

Reply:
I believe I understood what suggestions you were making so I've added another commit which pushes the conversion higher. Note the impact it has on the `HdbscanModel`. Again, a separate PR will be raised with these actual integration changes.

Reply:
Ok, sounds good.